### ISSUE TITLE
UnicodeError when attempting to access OBS

### ISSUE THREAD

**Issue by soltanianalytics** [opentelekomcloud/python-otcextensions#64]
When attempting to use this library to access files stored in OBS, the authentication appears to fail due to this Python bug: https://bugs.python.org/issue32958

It appears to me that the authentication method employed by this package fails because it tries to use the AK/SK pair as credentials, which are 60 characters, and the ToASCII method fails when attempting to parse the resulting url which contains the AK/SK pair in `ak:sk@www...` format (see also the answer to https://stackoverflow.com/questions/25103126/label-empty-or-too-long-python-urllib2).

my `clouds.yaml` looks like that:
```
clouds:
 otc:
   auth:
     username: 'username'
     password: 'password'
     project_name: 'eu-de'
     auth_url: 'https://iam.eu-de.otc.t-systems.com:443/v3'
     user_domain_name: 'OTC-EU-DE-0000000000100xxxxxxx'
   interface: 'public'
   identity_api_version: 3 # !Important
   ak: '20-digit AK' # AK/SK pair for access to OBS
   sk: '40-digit SK'
```

Assuming a configured `clouds.yaml`, the following should replicate the error:
```python
import openstack
from otcextensions import sdk

conn = openstack.connect(cloud='otc')
sdk.register_otc_extensions(conn)

# Error 1
conn.obs.get_container('bucker_name') # Variant one - leads to HttpException 405

# Error 2
for cont in conn.obs.containers():
 a_container = cont # the last container in the generator will end up as a_container
conn.obs.get_object('file_name.txt', container=a_container) # this will lead to UnicodeError: label empty or too long

# Error 3 - same as error 2
conn.obs.get_object('file_name.txt', container='bucket_name') # this will lead to the same error as Error 2
```

Errors 2 and 3:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/encodings/idna.py", line 165, in encode
    raise UnicodeError("label empty or too long")
UnicodeError: label empty or too long

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/otcextensions/sdk/obs/v1/_proxy.py", line 238, in get_object
    requests_auth=self._get_req_auth(endpoint))
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/openstack/proxy.py", line 46, in check
    return method(self, expected, actual, *args, **kwargs)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/otcextensions/sdk/sdk_proxy.py", line 197, in _get
    requests_auth=requests_auth,
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/otcextensions/sdk/sdk_resource.py", line 174, in get
    response = session.get(request.url, **get_args)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/keystoneauth1/adapter.py", line 386, in get
    return self.request(url, 'GET', **kwargs)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/openstack/proxy.py", line 98, in request
    **kwargs)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/keystoneauth1/adapter.py", line 248, in request
    return self.session.request(url, method, **kwargs)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/keystoneauth1/session.py", line 913, in request
    resp = send(**kwargs)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/keystoneauth1/session.py", line 1004, in _send_request
    resp = self.session.request(method, url, **kwargs)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/urllib3/connectionpool.py", line 672, in urlopen
    chunked=chunked,
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/urllib3/connectionpool.py", line 376, in _make_request
    self._validate_conn(conn)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/urllib3/connectionpool.py", line 994, in _validate_conn
    conn.connect()
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/urllib3/connection.py", line 300, in connect
    conn = self._new_conn()
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/urllib3/connection.py", line 157, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/urllib3/util/connection.py", line 61, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/local/lib/python3.7/socket.py", line 748, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
UnicodeError: encoding with 'idna' codec failed (UnicodeError: label empty or too long)
```

Error 1 - I'm just assuming that it's due to the same underlying problem, but I don't know that
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/otcextensions/sdk/obs/v1/_proxy.py", line 89, in get_container
    return self._head(_container.Container, container)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/otcextensions/sdk/sdk_proxy.py", line 262, in _head
    requests_auth=requests_auth,
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/otcextensions/sdk/sdk_resource.py", line 209, in head
    self._translate_response(response, has_body=False)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/otcextensions/sdk/obs/v1/container.py", line 55, in _translate_response
    exceptions.raise_from_response(response, error_message=response.text)
  File "/home/bijan/dev/clients/geton/tests/env/lib/python3.7/site-packages/openstack/exceptions.py", line 229, in raise_from_response
    http_status=http_status, request_id=request_id
openstack.exceptions.HttpException: HttpException: 405: Client Error for url: https://obs.eu-de.otc.t-systems.com/, Method Not Allowed
```

I would suggest changing the authentication flow. In any case, is there a known workaround?

**Comment by Nils-Magnus** [comment]
Thanks for reporting. We're currently triaging. For the sake of completeness, could you please add
```python
openstack.enable_logging(debug=True)
```
to your script and paste the `REQ` and `RESP` fields?

**Comment by soltanianalytics** [comment]
We abandoned Telekom Cloud at the time and I don't have the script anymore, but are considering it again for another project. Has this issue been investigated and fixed?

**Comment by anton-sidelnikov** [comment]
ak/sk not contained in url, it stored in headers. Will add some functional tests for OBS.

**Comment by anton-sidelnikov** [comment]
fixed in #288
